runtime/src/deterministic.rs

```diff
@@ -227,15 +227,59 @@ pub struct Executor {
     time: Mutex<SystemTime>,
     tasks: Arc<Tasks>,
     sleeping: Mutex<BinaryHeap<Alarm>>,
-    partitions: Mutex<HashMap<String, Partition>>,
     shutdown: Mutex<Stopper>,
     finished: Mutex<bool>,
-    recovered: Mutex<bool>,
 }
 
+impl Executor {
+    /// Clear all tasks.
+    ///
+    /// We don't consume [Executor] because it must remain upgradable for any tasks
+    /// that interact with the runtime during drop.
+    fn stop(&self) {
```

Comment:
We now forcibly drop all running tasks when `start` finishes (like `tokio`).

If we implement `Drop` handlers for all stateful functions, this obviates the need to track shutdown handling 👀